lib/jobs/ucs-discovery.js

```diff
@@ -140,44 +140,57 @@ function UcsDiscoveryJobFactory(
                         return res.body;
                     })
                     .map(function(data) {
-                        assert.array(data.macs);
-                        assert.string(data.name);
-                        assert.string(data.path);
-
-                        var config = Object.assign({}, self.settings);
-                        obm = {
-                            config: config,
-                            service: 'ucs-obm-service'
-                        };
-                        var node = {
-                            type: 'compute',
-                            name: data.path,
-                            identifiers: data.macs,
-                            obm:[obm],
-                            relations : []
-                        };
-
-                        // Update existing node  or create a new one
-                        return waterline.nodes.needOne({identifiers:data.macs})
-                            .then(function(curNode) {
-                                return waterline.nodes.updateOne(
-                                    { id: curNode.id },
-                                    node
-                                )
-                                .then(function(data){
-                                    return data;
+                        self.createNode(data, 'compute');
+                    });
+            }else{
+                logger.warning('No rackmount servers found Found');
+                return;
+            }
+        });
+    };
+
+    /**
+     * @function createServers
+     * @description discovers all the chassis servers in the Cisco UCS
+     */
+    UcsDiscoveryJob.prototype.createServers = function (root) {
+        var self = this;
+        var username = this.settings.username;
+        var password = this.settings.password;
+        var ucs = this.settings.ucs;
+        var baseurl = this.settings.protocol+"://"+this.settings.host+ ":"+this.settings.port;
+        var url= baseurl + "/chassis?host=" + ucs+ "&user="+ username + "&password=" + password;/* jshint ignore:line */
+        return Promise.try(function () {
+            if (_.has(root, 'Chassis')) {
+                return self.ucs.clientRequest(url)
+                    .then(function(res) {
+                        assert.object(res);
+                        return res.body;
+                    })
+                    .map(function(chassisData) {
+                        return self.createNode(chassisData, 'enclosure')
```

Comment:
Please change node type to use the constant that defined at https://github.com/RackHD/on-core/blob/master/lib/common/constants.js#L328